packages/api/cms-api/src/user-permissions/user-permissions.service.ts

```diff
@@ -0,0 +1,148 @@
+import { ContentScope } from "./interfaces/content-scope.interface";
+import { Permission } from "./interfaces/user-permission.interface";
+import { USER_PERMISSIONS_CONFIG_SERVICE } from "./user-permissions.const";
+import { UserPermissionConfigInterface, UserPermissions } from "./user-permissions.types";
+
+@Injectable()
+export class UserPermissionsService {
+    constructor(
+        @Inject(USER_PERMISSIONS_CONFIG_SERVICE) private readonly service: UserPermissionConfigInterface,
+        @InjectRepository(UserPermission) private readonly permissionRepository: EntityRepository<UserPermission>,
+        @InjectRepository(UserContentScopes) private readonly contentScopeRepository: EntityRepository<UserContentScopes>,
+    ) {}
+
+    async getAvailableContentScopes(): Promise<ContentScope[]> {
+        return this.service.getAvailableContentScopes ? this.service.getAvailableContentScopes() : [];
+    }
+
+    async getAvailablePermissions(): Promise<(keyof Permission)[]> {
+        return [
+            ...new Set<keyof Permission>([
+                "dam",
+                "pageTree",
+                "userPermissions",
+                "system",
+                ...(this.service.getAvailablePermissions ? await this.service.getAvailablePermissions() : []),
+            ]),
+        ];
+    }
+
+    async getUser(id: string): Promise<User> {
+        return this.service.getUser(id);
+    }
+
+    async findUsers(args: FindUsersArgs): Promise<[User[], number]> {
+        return this.service.findUsers(args);
+    }
+
+    async checkContentScopes(contentScopes: ContentScope[]): Promise<void> {
+        const availableContentScopes = await this.getAvailableContentScopes();
+        contentScopes.forEach((scope) => {
+            if (!availableContentScopes.some((cs) => isEqual(cs, scope))) {
+                throw new Error(`ContentScope does not exist: ${JSON.stringify(scope)}.`);
+            }
+        });
+    }
+
+    async getPermissions(userId: string): Promise<UserPermission[]> {
+        const availablePermissions = await this.getAvailablePermissions();
+        const permissions = (
+            await this.permissionRepository.find({
+                $and: [{ userId }, { permission: { $in: availablePermissions } }],
+            })
+        ).map((p) => {
+            p.source = UserPermissionSource.MANUAL;
+            return p;
+        });
+        if (this.service.getPermissionsForUser) {
+            const user = await this.getUser(userId);
+            if (user) {
+                let permissionsByRule = await this.service.getPermissionsForUser(user);
+                if (permissionsByRule === UserPermissions.allPermissions) {
+                    permissionsByRule = availablePermissions.map((permission) => ({ permission }));
+                }
+                for (const p of permissionsByRule) {
+                    const permission = new UserPermission();
+                    permission.id = getUuid(JSON.stringify(p));
```

Comment:
this results in a non-table id.

do we need an id at all?

Reply:
I actually had a version without this but it makes the admin-part much more complicated. Since the id is only used for the admin and not persisted it shouldn't be a problem that it's not stable.